core/ledger/kvledger/txmgmt/pvtstatepurgemgmt/purge_mgr_test.go

```diff
@@ -163,7 +159,7 @@ func testPurgeMgrForCommittingPvtDataOfOldBlocks(t *testing.T, dbEnv privacyenab
 }
 
 func TestKeyUpdateBeforeExpiryBlock(t *testing.T) {
-	dbEnv := &privacyenabledstate.LevelDBCommonStorageTestEnv{}
+	dbEnv := testEnvs[0]
```

Comment:
This seems a little fragile, what do we gain by pulling it from the array? If we were to support another storage engine in the future, if it was added to the start of testEnvs instead of the end, it would break the expected outcome here.

Reply:
thanks for the review. very good point, it's definitely very fragile. 
The intent here is that couchDB external container is only being started once for the entire package (doing so by calling _testEnv.Init_ at the beginning of every test), and **TestMain** will loop through _testEnvs_ after all tests finished in order to cleanup couchDB. If we declare new test environments in specific test, **TestMain** will not cleanup couchDB properly. 
Open for any suggestions.

Reply:
Ahh, off the top of my head, you could create a map (which you can still iterate over), but here you could reference by its key, `testEnvs["level"]`

Reply:
Agree; map is preferred.

Reply:
Yup, this map based approach is already being used in some other package as well - https://github.com/hyperledger/fabric/blob/d80c5f5a60d59a78891e1c958018a4bfeb892182/core/ledger/kvledger/txmgmt/txmgr/lockbasedtxmgr/pkg_test.go#L65.

Reply:
changes applied accordingly. thanks